### PR TITLE
Optional attendees

### PR DESCRIPTION
**Pull request by zoebarth**
Add test cases as described in the exercises to account for optional attendees. Write algorithm to account for optional attendees as described below:

"if one or more time slots exists so that both mandatory and optional attendees can attend, return those time slots. Otherwise, return the time slots that fit just the mandatory attendees."

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -63,9 +75,35 @@
...
+        }
+      }
+
+      // Add available time after all evenets.
+      if (start + meetingDuration <= TimeRange.END_OF_DAY) {
+        available.add(TimeRange.fromStartEnd(start, TimeRange.END_OF_DAY, true));
+      }
+      
+      return available;
+      }
```
Comment (genedwards) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:106]: nit: indentation is off

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -63,9 +75,35 @@
...
+        // Check end time to account for when an event has an earlier start time but later end time.
+        if (time.end() > start) {
+          start = time.end();
+        }
+      }
+
+      // Add available time after all evenets.
+      if (start + meetingDuration <= TimeRange.END_OF_DAY) {
+        available.add(TimeRange.fromStartEnd(start, TimeRange.END_OF_DAY, true));
+      }
```
Comment (genedwards) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:103]: This looks like the exact same thing we did for the optionalUnavailable. Normally, if it's only repeated twice, I don't think pulling to a separate function is needed, but it also has enough complexity that pulling these out to a separate function would be helpful.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -16,44 +16,58 @@
 
 import java.util.*;
```
Comment (mecolmg) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:17]: Don't add wildcard imports.
https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (zoebarth) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Fixed.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (zoebarth) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Done.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (zoebarth) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Done.

**Pull request merged**